modules/developers-guide/pages/cli-reference/dfx-canister.adoc

```diff
@@ -669,6 +674,124 @@ Set "r7inp-6aaaa-aaaaa-aaabq-cai" as controller of "open_sf_assets".
+
+[width="100%",cols="<32%,<68%",options="header"]
+|===
+|Flag |Description
+|`+-h+`, `+--help+` |Displays usage information.
+|`+--query+` |Sends a query request to a canister.
+|`+--update+` |Sends an update request to the canister. This is the defualt method if the `+--query+` method is not used.
+|`+-V+`, `+--version+` |Displays version information.
+|===
+
+=== Options
+
+You can specify the following options for the `+dfx canister sign+` command.
+
+[width="100%",cols="<32%,<68%",options="header"]
+|===
+|Option |Description
+|`+--expire-after <expire-after>+` |Specifies how long will  will be valid before it expires and cannot be sent. Specify  in seconds. If not defined, the default is 300s (5m)
+|`+--file <file>+` |Specifies the output file name. The  default is `+message.json+`.
+|`+--random <random>+` |Specifies the configuration for generating random arguments.
+|`+--type <type>+` |Specifies the data type for the argument when making a call using an argument.
+
+By default, you can specify arguments using the link:../../candid-guide/candid-intro{outfilesuffix}[Candid] (`+idl+`) syntax for data values.
+For information about using Candid and its supported types, see link:../../candid-guide/candid-howto{outfilesuffix}#idl-syntax[Interact with a service in a terminal] and link:../../candid-guide/candid-types{outfilesuffix}[Supported types]. 
+You can use `+raw+` as the argument type if you want to pass raw bytes.
+|===
+
+=== Arguments
+
+You can specify the following arguments for the `+dfx canister sign+` command.
+
+[width="100%",cols="<32%,<68%",options="header"]
+
+|===
+|Argument |Description
+|`+canister_name+` |Specifies the name of the canister to call. The canister name is a required argument and should match the name you have configured for a project in the `+canisters+` section of the `+dfx.json+` configuration file.
+
+|`+method_name+` |Specifies the method name to call on the canister.
+The canister method is a required argument.
+
+|`+argument+` |Specifies the argument to pass to the method.
+Depending on your program logic, the argument can be a required or optional argument.
+You can specify a data format type using the `+--type+` option if you pass an argument to the canister.
+By default, you can specify arguments using the link:../../candid-guide/candid-intro{outfilesuffix}[Candid] (`+idl+`) syntax for data values. 
+For information about using Candid and its supported types, see link:../../candid-guide/candid-howto{outfilesuffix}#idl-syntax[Interact with a service in a terminal] and link:../../candid-guide/candid-types{outfilesuffix}[Supported types].
+You can use `+raw+` as the argument type if you want to pass raw bytes.
+|===
+
+=== Examples
+
+Use the `+dfx canister sign+` command to create a signed `method.json` file using the pricipal ID associated with the identitiy you created using the Privacy Enhanced Mail (PEM) file by running a command similar to the following:
```

Comment:
```suggestion
Use the `+dfx canister sign+` command to create a signed `message.json` file using the principal associated with the identity you created using the Privacy Enhanced Mail (PEM) file by running a command similar to the following:
```

Reply:
For now, I think we need to add --no-wallet to the example and explain 
1) this example connects to the network specified by the `ic` network alias and 
2) that `--no-wallet` is required to execute the command using the user Identity instead of a cycles wallet canister identifier.
Hopefully, number 2 goes away, but currently that is a required setting.

Reply:
in the SDK tutorials, `--no-wallet` is used with the wallet canister identifier...are you saying that I need to use the principal user identity? Like:
`dfx canister --no-wallet --network+ic sign --expire-after=1h **wokyb-oonsl-6wxtc-3bded-4rlgh-aaegd-fnow5-d4paa-s3gkc-cjxxh-lae** create nrurons '("PUBLIC_KEY")'`
instead of:
dfx canister --no-wallet --network=ic sign --expire-after=1h **rno2w-sqaaa-aaaaa-aaacq-cai** create_neurons ‘(“PUBLIC_KEY”)'` ?

Reply:
`--no-wallet `is used to make a `call` using the user principal identity instead of bypassing the user identity and making the call through the wallet identifier. 
In the tutorials you refer to, I think we were calling the wallet canister (so that identifier was an argument for dfx canister call) but using the user's identity (by adding the `--no-wallet` option).
In the example:
`dfx canister --no-wallet --network=ic sign --expire-after=1h rno2w-sqaaa-aaaaa-aaacq-cai create_neurons ‘(“PUBLIC_KEY”)' `
we are making a call to canister "`rno2w-sqaaa-aaaaa-aaacq-cai`" -- AKA the "`governance`" canister and the method `create_neurons`

And, yes, it ain't you. It is a bit confusing.